runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/ParDoTranslation.java

```diff
@@ -161,9 +158,40 @@ public boolean canTranslate(PTransform<?, ?> pTransform) {
   }
 
   public static ParDoPayload translateParDo(
-      final ParDo.MultiOutput<?, ?> parDo, Pipeline pipeline, SdkComponents components)
-      throws IOException {
+      AppliedPTransform<?, ?, ?> appliedPTransform, SdkComponents components) throws IOException {
```

Comment:
`AppliedPTransform<ParDo.MultiOutput<?, ?>, ? ,?>` or some such? Avoids the cast below. Might force a cast at the caller, and that makes sense because the caller is the one who is making the decision to pass the param.

Reply:
Done